src/address.pest

```diff
@@ -0,0 +1,21 @@
+// Basic Multiserver grammar.
+name = { ( ASCII_ALPHA_LOWER | ASCII_DIGIT | "-" ) + }
+// TODO: Implement escape character.
+special = { ( "!" | ";" | "~" ) }
+data = { (! (WHITE_SPACE | special ) ~ ANY ) + }
+protocol = { name ~ ":" ~ data*}
+address = { protocol ~ ("~" ~ protocol) * }
+multiaddress = { address ~ ( ";" ~ address ) * }
+
+// Protocol: net
```

Comment:
Maybe this would be better as its own grammar? The problem with the basic grammar is that we still have to extract the host and port, but since IPv6 addresses don't require brackets (e.g. `net::::8008` instead of `net:[::]:8008`) we have to parse out the host anyway.